src/controller/java/src/chip/devicecontroller/model/ClusterState.java

```diff
@@ -51,6 +55,33 @@ public Optional<Long> getDataVersion() {
     return dataVersion;
   }
 
+  /**
+   * Convenience utility for getting all attributes in Json string format.
+   *
+   * @return all attributes in Json string format., or empty string if not found.
+   */
+  public String getAttributesJson() {
+    JSONObject combinedObject = new JSONObject();
+    attributes.forEach(
```

Comment:
can this be shortened somehow? I see attributeId is not used anywhere, so probably just using values works somehow. Thinking of:

```kt
attributes.values
   .map { it.getJson() }
   .filter { it != null }
```

to grab all the json objects. After which we can iterate . Overall hard to write the function without a kotlin IDE (I don't have access to one right now) but this feels like it could be shortened.

Reply:
well, it is java, not koltin, attributes.values return collection, it seems we cannot apply map to collection.

Reply:
done, thanks